proxy/client_test.go

```diff
@@ -0,0 +1,213 @@
+package proxy
+
+import (
+	"bytes"
+	"io/ioutil"
+	"net/http"
+	"net/http/httptest"
+	"net/url"
+	"reflect"
+	"strings"
+	"testing"
+)
+
+func TestExtractNextBatch(t *testing.T) {
+	input := `{
+		"next_batch": "s361093_69_4_8353_1",
+		"presence": {
+			"events": [
+				"event"
```

Comment:
This is technically not valid, because event must not be a string.

I try to make test cases be actually valid data, because it would not be unreasonable for us to error if we failed to fully parse the response.